lib/classroom_api.rb

```diff
@@ -8,190 +8,229 @@ def initialize(access_token)
         @debug = true
         return @debug
       end
+      # check if database has rooms that are not in API anymore
+      if @rooms_in_db.present?
+        RoomContact.where(rmrecnbr: @rooms_in_db).delete_all
+        RoomCharacteristic.where(rmrecnbr: @rooms_in_db).delete_all
+        if Room.where(rmrecnbr: @rooms_in_db).delete_all
+          @log.api_logger.info "add_facility_id_to_classrooms, delete #{@rooms_in_db} room(s) from the database"
+        else
+          @log.api_logger.debug "add_facility_id_to_classrooms, error: could not delete records with #{@rooms_in_db} rmrecnbr"
+          @debug = true
+          return @debug
+        end
+      end
+    rescue StandardError => e
+      # example: Errno::ETIMEDOUT: Operation timed out - user specified timeout
+      @log.api_logger.debug "add_facility_id_to_classrooms, error: API return: #{e.message}"
+      @debug = true
     end
     return @debug
   end
 
 
   def get_classrooms_list
-    result = {'success' => false, 'errorcode' => '', 'error' => '', 'data' => {}}
-    url = URI("https://gw.api.it.umich.edu/um/aa/ClassroomList/Classrooms?BuildingID=1005046")
-
-    http = Net::HTTP.new(url.host, url.port)
-    http.use_ssl = true
-    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
-
-    request = Net::HTTP::Get.new(url)
-    request["x-ibm-client-id"] = "#{Rails.application.credentials.um_api[:buildings_client_id]}"
-    request["authorization"] = "Bearer #{@access_token}"
-    request["accept"] = 'application/json'
-
-    response = http.request(request)
-    response_json = JSON.parse(response.read_body)
-    if response_json['errorCode'].present?
-      result['errorcode'] = response_json['errorCode']
-      result['error'] = response_json['errorMessage']
-    else
-      result['success'] = true
-      result['data'] = response_json['Classrooms']['Classroom']
+    begin
+      result = {'success' => false, 'errorcode' => '', 'error' => '', 'data' => {}}
+      classrooms = []
+      start_index = 0
+      count = 1000
+      next_page = true
+      while next_page do
+        url = URI("https://gw.api.it.umich.edu/um/aa/ClassroomList/v2/Classrooms?$start_index=#{start_index}&$count=#{count}")
+
+        http = Net::HTTP.new(url.host, url.port)
+        http.use_ssl = true
+        http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+        request = Net::HTTP::Get.new(url)
+        request["x-ibm-client-id"] = "#{Rails.application.credentials.um_api[:buildings_client_id]}"
+        request["authorization"] = "Bearer #{@access_token}"
+        request["accept"] = 'application/json'
+
+        response = http.request(request)
+        link = response.to_hash["link"].to_s
+        if link.include? "rel=next"
+          start_index += count
+        else
+          next_page = false
+        end
+        response_json = JSON.parse(response.read_body)
+        if response.code == "200"
```

Comment:
Why? The approved Department API uses the same condition.

Reply:
Yes, you are correct that we approved the previous pull request with similar code. On second thought, I think we should consider creating a constant variable so that we don't have hardcoded occurrences in the code. Thank you.

Reply:
did it in [c8ca647](https://github.com/lsa-mis/mi_classrooms/pull/326/commits/c8ca64751f8a760058c4fc028b60aeebaef0ab6d).
To make you happy.